test_programs/compile_success_empty/arithmetic_generics/src/main.nr

```diff
@@ -101,8 +101,11 @@ fn demo_proof<let N: u32>() -> Equiv<W<(N * (N + 1))>, (Equiv<W<N>, (), W<N>, ()
     let p1: Equiv<W<(N + 1) * N>, (), W<N * (N + 1)>, ()> = mul_comm();
     let p2: Equiv<W<N * (N + 1)>, (), W<N * N + N>, ()> = mul_add::<N, N, 1>();
     let p3_sub: Equiv<W<N>, (), W<N>, ()> = mul_one_r();
-    let p3: Equiv<W<N * N + N>, (), W<N * N + N>, ()> = add_equiv_r::<N * N, N, N, _, _>(p3_sub);
-    equiv_trans(equiv_trans(p1, p2), p3)
+    let _p3: Equiv<W<N * N + N>, (), W<N * N + N>, ()> = add_equiv_r::<N * N, N, N, _, _>(p3_sub);
+    let _p1_to_2 = equiv_trans(p1, p2);
+
+    // equiv_trans(p1_to_2, p3)
+    std::mem::zeroed()
```

Comment:
@michaeljklein had to disable this final check. I'm getting a very lengthy error here with this PR now.